nilearn/decomposition/multi_pca.py

```diff
@@ -76,15 +85,19 @@ def session_pca(imgs, mask_img, parameters,
             confounds=confounds,
             copy=copy)
     if n_components <= data.shape[0] // 4:
-        U, S, _ = randomized_svd(data.T, n_components)
+        U, S, _ = cache(randomized_svd, memory, memory_level=memory_level,
+                        func_memory_level=2)(
+            data.T, n_components, random_state=random_state)
     else:
-        U, S, _ = linalg.svd(data.T, full_matrices=False)
+        U, S, _ = cache(linalg.svd, memory, memory_level=memory_level,
+                        func_memory_level=2)(
+            data.T, full_matrices=False)
```

Comment:
Ok i misunderstood the question. This is because linalg.svd is
deterministic.
On 11 Jul 2015 15:25, "bthirion" <notifications@github.com> wrote:

> In nilearn/decomposition/multi_pca.py
> <https://github.com/nilearn/nilearn/pull/630#discussion_r34412606>:
>
> >      else:
> > -        U, S, _ = linalg.svd(data.T, full_matrices=False)
> > +        U, S, _ = cache(linalg.svd, memory, memory_level=memory_level,
> > +                        func_memory_level=2)(
> > +            data.T, full_matrices=False)
>
> I think that @banilo <https://github.com/banilo> meant that there is no
> random_state given here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nilearn/nilearn/pull/630/files#r34412606>.
>